src/System.Text.Json/tests/JsonStringTests.cs

```diff
@@ -99,19 +99,18 @@ public static void TestGuid()
         public static void TestDateTime(DateTime dateTime)
         {
             var jsonString = new JsonString(dateTime);
-            Assert.Equal(dateTime.ToString("s", CultureInfo.InvariantCulture), jsonString.Value);
+            Assert.Equal(dateTime.ToString("O", CultureInfo.InvariantCulture).Replace(".0000000", string.Empty), jsonString.Value);
```

Comment:
This test should adapt the writer's trimming tests, just like the updated JsonObjectTests.

Reply:
Done, but the last CI check failed, for seemingly unrelated reasons 😢

Reply:
That CI failure definitely looks unrelated:

```
System.Net.Security.Tests.ClientAsyncAuthenticateTest.ClientAsyncAuthenticate_MismatchProtocols_Fails(serverProtocol: Tls, clientProtocol: Tls12, expectedException: typeof(System.IO.IOException)) [FAIL]
      Assert.IsAssignableFrom() Failure
      Expected: typeof(System.IO.IOException)
      Actual:   typeof(System.ObjectDisposedException)
      Stack Trace:
        /_/src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs(79,0): at System.Net.Security.Tests.ClientAsyncAuthenticateTest.ClientAsyncAuthenticate_MismatchProtocols_Fails(SslProtocols serverProtocol, SslProtocols clientProtocol, Type expectedException)
        --- End of stack trace from previous location ---
  Finished:    System.Net.Security.Tests
``` 
(looks like classic race condition) but the null ref here is a straight up infra bug so I filed https://github.com/dotnet/arcade/issues/4219 to track that.